rust/crd/src/lib.rs

```diff
@@ -98,54 +104,97 @@ pub struct SparkApplicationStatus {
+    /// Spark to create the executor pods.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub executor: Option<RoleGroup<RoleConfigFragment>>,
-    #[serde(default, skip_serializing_if = "Option::is_none")]
-    pub stopped: Option<bool>,
+
+    /// A map of key/value strings that will be passed directly to spark-submit.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub spark_conf: Option<HashMap<String, String>>,
+
+    /// Job dependencies: a list of python packages that will be installed via pip, a list of packages
+    /// or repositories that is passed directly to spark-submit, or a list of excluded packages
+    /// (also passed directly to spark-submit).
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub deps: Option<JobDependencies>,
+
+    /// Configure an S3 connection that the SparkApplication has access to.
+    /// Read more in the [Spark S3 usage guide](DOCS_BASE_URL_PLACEHOLDER/spark-k8s/usage-guide/s3).
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub s3connection: Option<S3ConnectionDef>,
+
+    /// Arguments passed directly to the job artifact.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub args: Option<Vec<String>>,
+
+    /// A list of volumes that can be made available to the job, driver or executors via their volume mounts.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub volumes: Option<Vec<Volume>>,
+
+    /// A list of environment variables that will be set in the job pod and the driver and executor
+    /// pod templates.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub env: Option<Vec<EnvVar>>,
+
+    /// The log file directory definition used by the Spark history server.
+    /// Currently only S3 buckets are supported.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub log_file_directory: Option<LogFileDirectorySpec>,
 }
 
 #[derive(Clone, Debug, Default, Deserialize, JsonSchema, PartialEq, Eq, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct JobDependencies {
+    /// Under the `requirements` you can specify Python dependencies that will be installed with `pip`.
+    /// Example: `tabulate==0.8.9`
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub requirements: Option<Vec<String>>,
+
+    /// A list of packages that is passed directly to `spark-submit`.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub packages: Option<Vec<String>>,
+
+    /// A list of repositories that is passed directly to `spark-submit`.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub repositories: Option<Vec<String>>,
+
+    /// A list of excluded packages that is passed directly to `spark-submit`.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub exclude_packages: Option<Vec<String>>,
```

Comment:
For all 4 fields: Please also make mandatory. it will default to an empty vec anyway

Reply:
See 27410e5.